### ISSUE TITLE
Azure deployment fails with template SAFE v2

### ISSUE THREAD

**Issue by Lewiszz** [SAFE-Stack/SAFE-template#414]
Is anyone else experiencing deployment issues on Azure with the current SAFE template or is something wrong with my Azure settings? I have been able to deploy the SAFE template (v2) before, without issues.

**Steps to reproduce:**
1. If necessary, run `dotnet new -i SAFE.Template`
2. Create a blank SAFE template with `dotnet new SAFE`
3. Restore using `dotnet tool restore`
4. Log in to Azure CLI with `az login`
5. Try to deploy with `fake built -t azure` 

**Problem:**
The following error occurs:
```
---------------------------------------------------------------------
Build Time Report
---------------------------------------------------------------------
Target          Duration
------          --------
Clean           00:00:00.0086400
InstallClient   00:00:25.4337813
Bundle          00:00:47.1892761
Azure           00:00:05.2795488   (ValidationError: Azure Error: InvalidTemplateDeployment
Message: The template deployment 'farmer-deploy-1693' is not valid according to the validation procedure. The tracking id is 'ebf3b1c9-c1ff-476c-9c09-cb460f36ab4c'. See inner 
errors for details.
Exception Details:
        Error Code: ValidationForResourceFailed
        Message: Validation failed for a resource. Check 'Error.Details[0]' for more information.
)
Total:          00:01:18.0774668
Status:         Failure
---------------------------------------------------------------------
Script reported an error:
-> BuildFailedException: Target 'Azure' failed.
-> One or more errors occurred. (ValidationError: Azure Error: InvalidTemplateDeployment
   Message: The template deployment 'farmer-deploy-1693' is not valid according to the validation procedure. The tracking id is 'ebf3b1c9-c1ff-476c-9c09-cb460f36ab4c'. See inner errors for details.
   Exception Details:
        Error Code: ValidationForResourceFailed
        Message: Validation failed for a resource. Check 'Error.Details[0]' for more information.
   )
-> ValidationError: Azure Error: InvalidTemplateDeployment
   Message: The template deployment 'farmer-deploy-1693' is not valid according to the validation procedure. The tracking id is 'ebf3b1c9-c1ff-476c-9c09-cb460f36ab4c'. See inner errors for details.
   Exception Details:
        Error Code: ValidationForResourceFailed
        Message: Validation failed for a resource. Check 'Error.Details[0]' for more information.
```

I have tried using different app names, but that didn't solve it so far. The Activity Log on Azure does log this ValidationError, but I don't see any detailed errors in the JSON attached there.

**Comment by Lewiszz** [comment]
Not sure if this is a duplicate of #411, since I get a different message (but might well be related to Farmer!).

**Comment by theimowski** [comment]
@isaacabraham

**Comment by isaacabraham** [comment]
@Lewiszz unlikely this is related to 411 as it successfully sent the template into Azure which then rejected it. I'm not sure that anything has changed - the only thing I can think of is that the app name is already taken - look in the build script, what's the name of the `webapp`? That name needs to be globally unique i.e. there can't be an app service from anyone, anywhere in the world, with that name.

**Comment by Lewiszz** [comment]
@isaacabraham Is there any other way to check whether a name exists? Because I tried many combinations, starting with ones that might have been used indeed, like "safe-v2-blank" and ""safe-v2-blank-test" etc, but I also tried variations on "blankv2safeii" and "lewisz21suez" for example and I can't imagine people already used all of them :) 

But if that's really the case then I should try inserting random keys maybe. Is there a max number of characters?
